### PR TITLE
Add new 'crs' coordinate variable when pyproj 2.0+ is installed

### PR DESCRIPTION
**Pull request by djhoese**
This adds a new 'crs' coordinate variable to all DataArrays that have an `AreaDefinition` defined. If the DataArray's area is a `SwathDefinition` then it adds lon/lat coordinate variables. For 2D arrays this means 2D uncomputed dask arrays will be added.

This PR also includes some other important changes:

1. The `abi_l1b` reader now drops the file 'x', 'y', 'x_image', and 'y_image' coordinate variables since they are either not used (x/y are overwritten by the base class) or are confusing to users (x_image/y_image). In the case of x_image/y_image, these coordinate variables aren't actually coordinates for the "Rad" netcdf variable, but xarray's handling of coordinates doesn't handle this.
2. The 'nucaps' reader now renames the "number_of_FORs" dimension to "y" to be more compatible with the rest of Satpy.
3. I moved a repeated chunk of code in many reader tests to a new function in `satpy.tests.utils`.

 - [x] Tests added and test suite added to parent suite 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [x] Fully documented

### CODE REVIEW

**Comment by djhoese** [comment]
@mraspaud should SwathDefinition-based DataArray's also include a "CRS" for `latlong` on WGS84? Or is that too confusing to handle in the future since you can't do `if 'crs' in data_arr.coords`? But it is more accurate.

**Comment by mraspaud** [comment]
About CRS for lon and lat, I think we should go for it, but we have to make sure in this case that x and y coordinates aren't mistaken for a lon and lat grid. That could be done by checking the units of x and y, which should then be different from degrees_north and degrees_east.

File: satpy/etc/readers/viirs_sdr.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ datasets:
     file_key: All_Data/{dataset_group}_All/Longitude
     units: "degrees"
```
Comment (mraspaud) [inline satpy/etc/readers/viirs_sdr.yaml]: This should be degrees_east, and latitudes should be degrees_north

File: satpy/etc/readers/viirs_sdr.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ datasets:
...
     standard_name: longitude
+    coordinates: [i_longitude, i_latitude]
```
Comment (mraspaud) [inline satpy/etc/readers/viirs_sdr.yaml]: Isn't this creating an infinite recursion ?

File: satpy/readers/yaml_reader.py [hunk 3]
```diff
@@ -718,6 +718,86 @@ def _load_dataset_area(self, dsid, file_handlers, coords):
...
+
+        """
+        if 'x' in data_arr.coords and 'y' in data_arr.coords:
+            return data_arr
+
+        if hasattr(area, 'get_proj_vectors'):
+            x, y = area.get_proj_vectors()
+        elif hasattr(area, 'get_proj_vectors_dask'):
+            # older pyresample with dask-only method
+            x, y = area.get_proj_vectors_dask(CHUNK_SIZE)
```
Comment (mraspaud) [inline satpy/readers/yaml_reader.py]: Should we take the opportunity to remove this call ?

**Comment by djhoese** [comment]
It seems most of the lack of coverage is coming from CRS (pyproj 2.0+) not being available. However, I have it installed locally and it works. I wonder if maybe I should have travis pip install it?

File: satpy/readers/yaml_reader.py (hunk 3 above)
Comment (djhoese) [reply satpy/readers/yaml_reader.py]: Done

File: satpy/etc/readers/viirs_sdr.yaml (hunk 2 above)
Comment (djhoese) [reply satpy/etc/readers/viirs_sdr.yaml]: Based on the way you implemented the stuff in the base reader, no. All the coordinates are collected for all datasets, then they are assigned.

File: satpy/etc/readers/viirs_sdr.yaml (hunk 1 above)
Comment (djhoese) [reply satpy/etc/readers/viirs_sdr.yaml]: Fixed.

**Comment by djhoese** [comment]
I think this PR is big enough that we don't want to change *all* readers to specify lon/lat coordinates for all of their geolocation. I can add a couple more if you'd like. I'm not sure how many polar readers we have.

**Pull request merged**